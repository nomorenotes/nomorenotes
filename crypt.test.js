const Hash = require("./crypt.js")

const jst = JSON.stringify.bind(JSON)
const salt = "$2a$10$yISxC8sqavt2TMvHPs8f7e" // Test salt, for reproduction purposes. DO NOT USE.

describe("crypt.js/Hash", () => {
    const texts = [
        "Hello World",
        "HelloWorld123",
        "llama",
        "",
        "\0\0\0\0\0\0\0\0"
    ]
    
    for (let text of texts) {
        describe(`from string ${jst(text)}`, () => {
            /** @type {Hash} */
            let hash
            beforeAll(async () => {
                hash = await Hash.hash(text, { salt }).then(h => (hash = h))
            })
            it(`should have been calculated before running tests`, async () => {
                expect(hash).toBeInstanceOf(Hash)
            })
            it(`should not have a different hash`, async () => {
                expect(hash.hash).toMatchSnapshot("hash " + jst(text))
            })
            it(`should compare to itself`, async () => {
                await expect(hash.compare(text)).resolves.toBeTruthy()
            })
            it(`should have the test salt and rounds`, async () => {
                expect(hash.salt).toBe(salt)
                expect(hash.rounds).toBe(10)
            })
            describe(`should not compare to different data`, () => {
                for (let text2 of texts) {
                    if (text === text2) continue
                    if ((text === "" && text2 === "\0\0\0\0\0\0\0\0") || (text2 === "" && text === "\0\0\0\0\0\0\0\0")) continue // null characters are ignored
                    test(jst(text2), async () => {
                        await expect(hash.compare(text2)).resolves.toBeFalsy()
                    })
                }
            })
        })
    }
})
