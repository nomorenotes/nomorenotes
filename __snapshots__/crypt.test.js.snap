// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`crypt.js/Hash from string "" should not have a different hash: hash "" 1`] = `"$2a$10$yISxC8sqavt2TMvHPs8f7emxTMgAaHp2L8zWMP.315oHn82cO8U3y"`;

exports[`crypt.js/Hash from string "\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000" should not have a different hash: hash "\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000" 1`] = `"$2a$10$yISxC8sqavt2TMvHPs8f7emxTMgAaHp2L8zWMP.315oHn82cO8U3y"`;

exports[`crypt.js/Hash from string "Hello World" should not have a different hash: hash "Hello World" 1`] = `"$2a$10$yISxC8sqavt2TMvHPs8f7eIkOvZLEXvluRPk57Zsau858WNJWtPlS"`;

exports[`crypt.js/Hash from string "HelloWorld123" should not have a different hash: hash "HelloWorld123" 1`] = `"$2a$10$yISxC8sqavt2TMvHPs8f7edVFaVmtS86ku7sb5F3OCSPVvSLlEmdS"`;

exports[`crypt.js/Hash from string "llama" should not have a different hash: hash "llama" 1`] = `"$2a$10$yISxC8sqavt2TMvHPs8f7ewV0mN3Uf78x0J4cGaGnUywed06oo86G"`;
